src/controllers/wallet/wallet.controller.js

```diff
@@ -510,7 +511,7 @@ async function sendTx(req, res) {
     const ret = { success: false, error: err.message };
     if (debug) {
       logger.debug('/send-tx failed', {
-        body: JSON.stringify(req.body),
+        body: JSONBigInt.stringify(req.body),
```

Comment:
This is a good question, I don't think it would create a problem per-se, but I haven't tested it.

However, consider the JSON section in mdn's BigInt [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#use_within_json):

> `JSON.stringify()` specifically leaves a backdoor for BigInt values: it would try to call the BigInt's `toJSON() `method. (It doesn't do so for any other primitive values.) Therefore, you can implement your own `toJSON() `method (which is one of the few cases where patching built-in objects is not explicitly discouraged).

Notice how it states that patching built-in objects is explicitly discouraged. You can find multiple other sources advocating against it. So, I wouldn't be too keen on patching `JSON.parse` or `JSON.stringify` directly, but we _could_ patch `BigInt.prototype.toJSON` as suggested in the docs. The second alternative, also explained there, is to use a `replacer`.

I ended up going with the latter because for the inverse case on `JSON.parse`, using a `reviver` is the only option. So to keep operations symmetric, I decided to use the same solution in both cases. Also, it makes things explicit rather than implicit.

> The reason I ask this is because in the future a dev may just use JSON without understanding the implications and introduce a bug that may only be caught if we use specific values during testing

I agree this could be problematic, but keep in mind that in the vast majority of cases the values will already be `BigInts`, even if they're small. I mean, whenever you're dealing with amounts, even if it's `1n`, it'll be a `BigInt`. The only case this would be a problem is when we have some object with a value that could be _either_ a `BigInt` or a `Number`, but I don't think we should have this case. Unfortunately this whole bigint deal is a bit of a pain in JS, and I think we should make an effort to keep this in mind whenever we're programming or reviewing. Later, using typing (either with TypeScript or just linters) will help prevent this kind of problem.

Reply:
The reviver we have only applies to API inputs, correct?
If we were to serialize/parse from anywhere else we would need to add this `toJSON` patch and use a reviver on the parse calls manually.

There can be a way to prevent `JSON.parse` without `reviver` as a custom eslint rule.
May be too much but checking `CallExpression:exit` where `JSON.parse(...)` only has 1 argument should be enough.

Reply:
I created an issue for this: https://github.com/HathorNetwork/hathor-wallet-headless/issues/511